betterproto/grpc/grpclib_client.py

```diff
@@ -0,0 +1,169 @@
+from abc import ABC
+import asyncio
+import grpclib.const
+from typing import (
+    Any,
+    AsyncIterable,
+    AsyncIterator,
+    Collection,
+    Iterable,
+    Mapping,
+    Optional,
+    Tuple,
+    TYPE_CHECKING,
+    Type,
+    Union,
+)
+from .._types import ST, T
+from .util.async_channel import AsyncChannel
+
+if TYPE_CHECKING:
+    from grpclib._protocols import IProtoMessage
+    from grpclib.client import Channel, Stream
+    from grpclib.metadata import Deadline
+
+
+_Value = Union[str, bytes]
+_MetadataLike = Union[Mapping[str, _Value], Collection[Tuple[str, _Value]]]
+_MessageSource = Union[Iterable["IProtoMessage"], AsyncIterable["IProtoMessage"]]
+
+
+class ServiceStub(ABC):
+    """
+    Base class for async gRPC clients.
```

Comment:
I think it's the "stub for the service", so ServiceStub seems fairly clear and I believe matches the fact that the other grpc libs (and this lib) generate `<ServiceName>Stub` subclasses.

Reply:
Ofcourse, it matches, yes, but we don't need to, and there might be good reasons not to:

For me the naming has been counter-intuitive from the beginning.

If the mission of Protobuf is to encapsulate the fact that these 'service calls' go over the wire, then it does sort of makes sense to call the Client `Service`. But then, the libs generate the Server as `Service` and the client as `ServiceStub`. To me that's very confusing.

Stub = unfinished piece of code. right?

- The Server code is basically _unfinished_, but its called `Service(Base)`

- The Client is actually completely functional when its generated, and does not need extension, but is called `ServiceStub`.

So to me it would have made much more sense the other way around. Server = `ServiceStub` and Client = `Service`. Such confusion is unnecessary friction in adoption of protobuf and betterproto. For many, betterproto may be the first entry point into protobuf, and their initial experience with it will be an important factor.

I personally would love to avoid the confusing terminology of Google Protobuf, and simply use `Client` and `Server`, which is what anyone will understand immediately.

Our mission is after all to create an easy-to-use (and understand) library for protobuf, and the names of the generated classes are not part of the protocol and format, so we can make them as user-friendly as we like.

Other opinions welcome 🦾

Reply:
TBH the name doesn't make much sense to me. 

From [the official grpc docs]

> On the client side, the client has a local object known as stub (for some languages, the preferred term is client) that implements the same methods as the service

I think *GRpcClient* would be a more relatable name in python. But I don't think it's worth changing it now because it would be a breaking change requiring a new major version.

Reply:
🥳  google gives us permission to call it client. that makes my day! 😛